aqsconverters/aq.py

```diff
@@ -37,16 +38,23 @@ def produce_annotation(self, aq_query_type, *args, **kwargs):
         print(f"\033[33mpatched {aq_query_type} with:\033[0m", args, kwargs)    
         print("\033[33mwriting annotation here:\033[0m", aq_module_name, args, kwargs)    
 
-        run = Run(uuid1())
+        run_id = uuid1()
+        run = Run(_id=run_id,
+                  name=str(run_id),
+                  type="Workflow")
         
-        aq_module = AstroqueryModule(_id="https://odahub.io/ontology#AQModule" + aq_module_name, name=aq_module_name)
+        aq_module = AstroqueryModule(_id="https://odahub.io/ontology#AQModule" + aq_module_name,
+                                     name=aq_module_name,
+                                     type="AstroqueryModule")
 
         #run.input_values = [AstrophysicalObject(_id=aq_module_name, name=aq_module_name)]
         run.isUsing = [aq_module]
 
         if aq_query_type == "query_object":
             obj_name = args[0]
-            obj =  AstrophysicalObject(_id="https://odahub.io/ontology#AstroObject" + obj_name.replace(" ","_"), name=obj_name) # normalize id
+            obj =  AstrophysicalObject(_id="https://odahub.io/ontology#AstroObject" + obj_name.replace(" ","_"),
+                                       name=obj_name,
+                                       type="AstrophysicalObject") # normalize id
```

Comment:
Seems like duplication, `AstrophysicalObject` class should indicate it has `type` `AstrophysicalObject`.

Reply:
and how could I extract it when querying the graph ?

Reply:
This sort of relation: between specific runs and concept of run should be taken care by the schema/framework.

And it is already done so, since each individual `run` is an instance of `Run` class.

For example see this piece of in `provenance.json`:

```json
{
    "@id": "51a97ba2-d5b0-11eb-a2dc-b5ff8b3b1162",
    "@type": [
      "http://odahub.io/ontology#Run"
    ],
```

Which looks also rdf turtle serialization as so:

```rdf
51a97ba2-d5b0-11eb-a2dc-b5ff8b3b1162 a <http://odahub.io/ontology#Run> .
```
Where `a` is `rdf:type` (https://www.w3.org/TR/turtle/)

Which is more or less what you wanted to achieve.